### ISSUE TITLE
Curl 8.5.0 bug in Alpine 3.19

### ISSUE THREAD

**Issue by leosamuele221** [nextcloud/all-in-one#4440]
### Steps to reproduce
1. Install nextcloud with latest label
2. Install "Memories" app and Memories external container

### Expected behavior 
Normal work of memories transcoder

### Actual behavior 
Nextcloud container can't connect to memories transcoder container
![Screenshot 2024-03-25 122745](https://github.com/nextcloud/all-in-one/assets/81746302/53d08089-a29c-4281-897e-9fc005cf0a3d)

![Screenshot 2024-03-25 123047](https://github.com/nextcloud/all-in-one/assets/81746302/86b4478b-d4db-45a9-b2d2-7af2e0d35530)

### Host OS 
CentOS 9

#### Nextcloud AIO version 
Nextcloud AIO v8.0.0

#### Current channel 
Latest

### Possible solution
Downgrade to Alpine 3.18
Upgrade to Alpine 3.20 (as soon as available)
Install curl 8.6.0 (see below)

#### Other valuable info 

The problem is curl 8.5.0 that is affected by https://github.com/curl/curl/issues/12558. This bug is actually solved in curl 8.6.0 so you just need to update curl.
![Screenshot 2024-03-25 123023](https://github.com/nextcloud/all-in-one/assets/81746302/c5595307-1b7e-497d-894f-639820bc7480)

Curl 8.6.0 isn't available from alpine package manager (in Alpine 3.19 the only version available is 8.5.0 https://pkgs.alpinelinux.org/packages?name=curl&branch=v3.19&repo=main) 
The only way to install curl 8.6.0 is to build it from source

To install from source follow this steps:

1. enter in nextcloud-aio-nextcloud container
`docker exec -it nextcloud-aio-nextcloud bash`

2. remove curl 8.5.0
`apk del curl`

3. download curl 8.6.0 (you can do it in any folder you want, personally I did it in "/home/test")
`wget https://github.com/curl/curl/releases/download/curl-8_6_0/curl-8.6.0.tar.gz`

4. unpack it
`tar -xf curl-8.6.0.tar.gz`

5. move to it
`cd curl-8.6.0`

6. install dependecies
`apk add openssl-dev g++ make autoconf libpsl-dev`

7. configure
`./configure --with-openssl`

8. make and install
`make && make install`

![image](https://github.com/nextcloud/all-in-one/assets/81746302/b42d772e-33b2-4303-a369-3f1669a10fab)

With curl 8.6.0 the problem is solved! 

![image](https://github.com/nextcloud/all-in-one/assets/81746302/767e3cf0-6c20-4ec7-a68c-2ec8ae4cd3ba)

If you want more info on how to build curl here is the docs https://github.com/curl/curl/blob/master/docs/INSTALL.md#unix

**Comment by Zoey2936** [comment]
https://gitlab.alpinelinux.org/alpine/aports/-/issues/15690 can you try running `apk upgrade --no-cache -a` inside the nextcloud container and then retry with the preinstalled curl?

**Comment by leosamuele221** [comment]
Tried with no luck
![image](https://github.com/nextcloud/all-in-one/assets/81746302/0ff2f43e-f0f4-4085-84d3-03aafae07e21)

(I did `cd home` to make sure I had reset the container)

**Comment by Zoey2936** [comment]
can you check if my prebuilt static curl binaries work? https://github.com/ZoeyVid/curl-quic/releases/latest

**Comment by leosamuele221** [comment]
For some reason it works from CLI
![Screenshot 2024-03-25 142758](https://github.com/nextcloud/all-in-one/assets/81746302/b58629da-7ea3-48d4-994f-f12c676f96c0)

But not from nextcloud
![image](https://github.com/nextcloud/all-in-one/assets/81746302/0dd4cb27-0593-4838-a9e9-ad91dcdc320d)

I also tried to execute it as www-data and it worked (obviously I change curl exec to 777) from CLI but not from nextcloud
![image](https://github.com/nextcloud/all-in-one/assets/81746302/0efeb591-3b52-4475-a42b-6cf0708d51a5)

**Comment by Zoey2936** [comment]
yes, because nextcloud trys to use libcurl

**Comment by Zoey2936** [comment]
can you try `apk add curl libcurl --repository http://dl-cdn.alpinelinux.org/alpine/edge/main` inside the nextcloud container and retry?

**Comment by leosamuele221** [comment]
Still nothing...
![image](https://github.com/nextcloud/all-in-one/assets/81746302/c96f98a7-0f7a-4275-a503-a1d8e4570f6c)

**Comment by leosamuele221** [comment]
The problem is in c-ares 1.24.0 https://github.com/networkservicemesh/deployments-k8s/issues/11185. This issue was resolved in c-ares 1.26.0

curl 8.6.0 from edge repository uses c-ares 1.24.0 so the problem still persist. 
When you build curl from source even if it's 8.6.0 it doesn't use c-ares, nor your build of curl uses c-ares @Zoey2936 that's why it worked

**Comment by leosamuele221** [comment]
I did `apk add curl libcurl --repository http://dl-cdn.alpinelinux.org/alpine/edge/main` and `apk upgrade c-ares` installing c-ares 1.27.0. But still doesn't work.
Now i'm confused

![image](https://github.com/nextcloud/all-in-one/assets/81746302/4fb4dc12-ece3-468e-a3c9-8127b1a65cf9)

**Comment by leosamuele221** [comment]
I made a script to automate build https://gist.github.com/leosamuele221/aed36a3b9135032b6a8ffe7274e58ce7

**Comment by Zoey2936** [comment]
can you report this here: https://gitlab.alpinelinux.org/alpine/aports/-/issues

**Comment by leosamuele221** [comment]
Yes i can, I'll do it as soon as possible.
But until the problem isn't solved somebody should add a good build of curl in the nextcloud-aio-nextcloud container as a patch

One option is to run this script https://gist.github.com/leosamuele221/aed36a3b9135032b6a8ffe7274e58ce7 when creating nextcloud container (maybe in the [Dockerfile](https://github.com/nextcloud/all-in-one/blob/main/Containers/nextcloud/Dockerfile)) so it builds a working curl

Another better option is to build a working curl and download it from github when creating the container

**Comment by szaimen** [comment]
Hi, can you update to v8.1.0 beta and check if that improves things?

**Comment by leosamuele221** [comment]
Hi, yes i updated it to 8.1.0
The problem is still there
Obv is a curl problem, not a concern of nextcloud.
![image](https://github.com/nextcloud/all-in-one/assets/81746302/3771f8b8-4b20-4e9d-bcfb-2dc200caaefb)

**Comment by leosamuele221** [comment]
![image](https://github.com/nextcloud/all-in-one/assets/81746302/cb532cba-d901-4738-ae41-da445864f172)

Problem is still here, even after updating to Alpine 3.19

**Comment by szaimen** [comment]
> can you report this here: https://gitlab.alpinelinux.org/alpine/aports/-/issues

Have you done so?

**Comment by leosamuele221** [comment]
do `curl -V`

**Comment by szaimen** [comment]
![image](https://github.com/nextcloud/all-in-one/assets/42591237/a9c5c8ff-e840-4b38-bc6b-6b64211c3ca4)

**Comment by phroggar** [comment]
I'm coming here periodically, because we have noticed a similar issue in our setup. 

I just want to share a solution which was working for us:

**Specify a hostname for the container you are unable to resolve internally: e.g. setting "--hostname=mycontainer.domain.local"** 

You might need to take all containers down and restart them, to have them pickup the change in the local network.

This is not brilliant and i wish things would be resolved on the CURL or C-ARES side of things - but i just upgraded to Alpine 3.20 (curl 8.7.1 (aarch64-alpine-linux-musl) libcurl/8.7.1 ... c-ares/1.28.1) and i still see the problematic behaviour (everything (ping, wget, etc.) is resolving fine, but curl is not able to resolve if the hostname is missing)

**Comment by szaimen** [comment]
All right, that is annoying. Can someone please log an issue [upstream](https://gitlab.alpinelinux.org/alpine/aports/-/issues)?

**Comment by AvivAvital2** [comment]
Encountered the issue while running alpine 3.18.
Initially upgraded from curl 8.5.0 to 8.7.1, but that alone did not resolve the issue for me.
Was able to resolve the issue by disabling c-ares, libidn2 and libpsl by using
`LDFLAGS="-static" PKG_CONFIG="pkg-config --static" ./configure --disable-shared --enable-static --with-openssl`

```
# curl --version
curl 8.7.1 (x86_64-pc-linux-musl) libcurl/8.7.1 OpenSSL/3.1.4 zlib/1.3.1 nghttp2/1.58.0
Release-Date: 2024-03-27
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns mqtt pop3 pop3s rtsp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS HSTS HTTP2 HTTPS-proxy IPv6 Largefile libz NTLM SSL threadsafe TLS-SRP UnixSockets
```

Hope this helps

**Comment by leosamuele221** [comment]
I reinstalled CentOS, a complete reset, reinstalled fresh nextcloud-aio and now everything works.

**Comment by szaimen** [comment]
This should hopefully be resolved with https://github.com/nextcloud/all-in-one/pull/4857

**Comment by szaimen** [comment]
> This should hopefully be resolved with #4857

This is now released with v9.1.0 Beta. Testing and feedback is welcome! See https://github.com/nextcloud/all-in-one#how-to-switch-the-channel
